### PR TITLE
refactor: convert APU page to React

### PR DESCRIPTION
**Pull request by ErickSharp**
Co-authored-by: Brandon Cabael <28613690+brandoncabael@users.noreply.github.com>

## Summary of Changes

Converts APU page to React. 

- [x] APU generator indication now has three modes with their respective visual representations (OFF, ON, STANDBY).
- [x] Correct APU generator indication visuals. 
- [x] Improved flashing animation found on ENGINE and APU page.
- [x] Correct positioning of elements in BLEED section.
- [x] New condition that makes BLEED crossbar turn amber.
- [x] Correct APU N Gauge behavior and visual representataion.
- [x] Correct EGT Gauge behavior and visual representation.

## Known Issues
Due to lack of simvar values, certain conditions are not yet implemented. This means that for now, the 'LOW OIL LEVEL' indication is not yet implemented, and the APU GEN and BLEED values will not be expressed in amber crosses yet in cases of data unavailability.

## Screenshots (if necessary)

BEFORE:
image
image

AFTER:
image
image
image
image

## References

https://www.youtube.com/watch?v=20QQ_JzpZo8&t=198s

## Additional context

Discord username (if different from GitHub): Erick [Z-6]#6484

## Testing instructions

* Check that 'APU GEN' is in amber if it is in OFF mode or the generator frequency is not between 390 - 410 Hz, or generator voltage is not between 110-120 V or the generator is overloaded.
* Generator load rises in 1V increments and is expressed in green unless it rises into overload. 
* Generator voltage rises in 1V increments and is expressed in green when 110 < V < 120 V or in amber otherwise.
* Generator frequency rises in 1Hz increments and is expressed in green when 390 < F < 410 Hz or in amber otherwise.
* Bleed value rises in 2PSI increments. 
* FUEL LO PR shown in amber in the memo region if there is low fuel pressure.
* FLAP OPEN shown in green when the flap is fully open, and in green pulsing when flap not fully closed and APU master switch push button is OFF and confirmed for 3 minutes.
* New amber mark on APU N gauge at 102%
* Valve is in amber and closed when the bleed air valve is closed and the APU Bleed is selected on the overhead for a minimum of 10 seconds.

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by ErickSharp** [comment]
> This is missing quite some functionality one could find in the original APU page. For example, the following TODO was added:
> 
> > There should be XX in amber in center of circle if air valve status info is not available
> 
> This functionality is found in the original code:
> 
> ```js
> // ADIRS1 on NAV is the normal operation situation.
> // Komp: If you switch the displays to DMC 3, then ADIRU 3 is the one providing the data.
> // Same when you select the air data selector to CAPT ON 3.
> // We'll wait for further implementation of those systems before making that functionality fully correct.
> const adirs1OnNav = SimVar.GetSimVarValue('L:A32NX_OVHD_ADIRS_IR_1_MODE_SELECTOR_KNOB', 'Enum') === 1;
> if (adirs1OnNav) {
>     this.APUBleedPressure.textContent = SimVar.GetSimVarValue("L:APU_BLEED_PRESSURE","PSI");
>     this.APUBleedPressure.setAttribute("class", "APUGenParamValue");
> } else {
>     this.APUBleedPressure.textContent = "XX";
>     this.APUBleedPressure.setAttribute("class", "APUGenParamValueWarn");
> }
> ```
> 
> The new code no longer uses variables such as `A32NX_ELEC_APU_GEN_1_LOAD_NORMAL`, but instead choses to replicate that logic within the page itself. Now that might be an okay choice and one can have a discussion about that, but as the writing of `A32NX_ELEC_APU_GEN_1_LOAD_NORMAL` and other variables wasn't removed from Rust code, I'm assuming it was not taken into account. It should be noted that the same pattern is also found on the ELEC page, so if we want to change how this works, then we should probably change it in both locations. Can you kick off a discussion on Discord? 😃
> 
> A FIXME like this is unclear to me. Does this mean the buttons are not broken?
> 
> > FIXME: and apu master switch is valid and apu gen push buttion is valid


Hello, David. I have incorporated your changes. Whenever you have time, feel free to take another look at the code and check to see if there are anymore inconsistencies or missing pieces : )

**Comment by ErickSharp** [comment]
> > It seems that most of our issues are resolved except for the size of the bottom gauge. Currently, the Arc component which we rely on only permits the drawing of semicircular arcs. I will make the necessary changes, but for now will mark this as a draft.
> 
> You might want to take a look at my cabin pressure PR as I have resolved this issue with respect to drawing gauges

Will do, thanks so much!

**Comment by davidwalschots** [comment]
Any progress on this?

**Comment by RichardPilbery** [comment]
> Any progress on this?

I think @ErickSharp was waiting on my PR, which has a modified gauge component. This is nearly ready for QA review (#5705)

**Comment by ErickSharp** [comment]
Getting this across the finish line today now that #5705 has been merged.

File: src/instruments/src/SD/Pages/Apu/Apu.tsx [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+    const [currentApuGenState, setCurrentApuGenState] = useState(apuGenState.STANDBY);
+
+    const [apuMasterPbOn] = useSimVar('L:A32NX_OVHD_APU_MASTER_SW_PB_IS_ON', 'Bool', 500);
+    const [apuGenPbOn] = useSimVar('A:APU GENERATOR SWITCH', 'Boolean', 750);
+    const [apuAvail] = useSimVar('L:A32NX_OVHD_APU_START_PB_IS_AVAILABLE', 'Bool', 1000);
+
+    useEffect(() => {
+        if (!apuMasterPbOn && !apuAvail /* FIXME: and apu master switch is valid and apu gen push buttion is valid */) {
+            setCurrentApuGenState(apuGenState.STANDBY);
+        } else if (!(currentApuGenState === apuGenState.STANDBY) && !apuGenPbOn /* FIXME: and apu master switch valid and apu gen pb valid */) {
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Apu/Apu.tsx:79]: **thought**: I would remove these `FIXME` comments. As they're describing a level of detail we're not close to achieving any time soon. These comments are saying: if we want to have an even higher level of detail we need to do something here. The problem is that such comments should probably go on every other line in our code base. Let's not do that 😀.

File: src/instruments/src/SD/Pages/Apu/Apu.tsx [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+                </text>
+
+                {currentApuGenState === apuGenState.OFF
+                    && <text x={50} y={70} className="Center FontNormal">OFF</text>}
+
+                {currentApuGenState === apuGenState.ON
+                    && (
+                        <>
+                            <SvgGroup x={60} y={55}>
+                                {/* FIXME: replaced by amber crosses when information is not available */}
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Apu/Apu.tsx:115]: **thought**: These "amber crosses" FIXME comments are more realistic to achieve soon, as we already have the means of writing ARINC429 data and thus could implement this later.

**general remark**: `FIXME` is a bit misnamed, as there's nothing wrong with the implementation: there are no bugs, just missing features. I'd write `TODO`. I'd also like to note that `TODO`s should be added sparingly, and they should describe exactly what needs to be done, to ensure it is clear for anyone else in the future. E.g. "Add amber crosses when information is not available, by implementing component _x_ and passing data from the APU's ECB in ARINC 429 format".

File: src/instruments/src/SD/Pages/Apu/Apu.tsx [hunk 3]
```diff
@@ -0,0 +1,521 @@
...
+        if (apuEgt.value > apuEgtCaution.value && apuEgt.value <= apuEgtWarning.value) {
+            setApuEgtIndicationColor('Amber');
+        }
+        if (apuEgt.value > apuEgtWarning.value) {
+            setApuEgtIndicationColor('Red');
+        }
+        // TODO FIXME: ecbAdvisoryDetected must be defined in a system later on.
+        // if (apuEgt.value <= apuEgtCaution.value && apuAvail && ecbAdvisoryDetected) {
+        //     setApuEgtIndicationColor('Pulse');
+        // }
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Apu/Apu.tsx:352]: **thought**: TODO or FIXME?
**suggestion**: Avoid checking in commented out code.

File: src/instruments/src/SD/Pages/Apu/Apu.tsx [hunk 4]
```diff
@@ -0,0 +1,521 @@
...
+            <SvgGroup x={x} y={y}>
+                {lowFuelPressure.value
+                    && <text className="Amber FontNormal" x={0} y={0}>FUEL LO PR</text>}
+
+                {apuFlapOpenPercentage === 100
+                    && <text className={`Green FontNormal ${isIntakeIndicationFlashing && 'FillPulse'}`} x={0} y={60}>FLAP OPEN</text>}
+
+                {/* FIXME: APU LOW OIL IS CURRENTLY NOT A VALUE */}
+                {/* { apuLowOil
+ && <text className="Green FillPulse" x={0} y={120}>LOW OIL LEVEL</text> } */}
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Apu/Apu.tsx:515]:
```suggestion
                {/* TODO: LOW OIL LEVEL message should be added by including oil quantity simulation in the APU and communicating that quantity or the warning itself through the ECB */}
```

File: src/instruments/src/SD/Pages/Apu/Apu.tsx [hunk 5]
```diff
@@ -0,0 +1,521 @@
...
+
+                <SvgGroup x={80} y={13}>
+                    <text x={0} y={0} className="White Center FontNormal">EGT</text>
+                    <text x={0} y={30} className="Cyan Center FontNormal" style={{ letterSpacing: -2 }}>&deg;C</text>
+                </SvgGroup>
+
+                <text
+                    x={10}
+                    y={80}
+                    // eslint-disable-next-line no-nested-ternary
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Apu/Apu.tsx:470]: **issue**: Breaks a linting rule.
**suggestion**: Fix the code to not break the rule.
```suggestion
```

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 1 above)
Comment (davidwalschots) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:79]: **suggestion**: Use `!==` instead of `!(===)`
```suggestion
        if (!apuMasterPbOn && !apuAvail /* FIXME: and apu master switch is valid and apu gen push buttion is valid */) {
            setCurrentApuGenState(apuGenState.STANDBY);
        } else if (currentApuGenState !== apuGenState.STANDBY && !apuGenPbOn /* FIXME: and apu master switch valid and apu gen pb valid */) {
```

File: src/instruments/src/SD/Pages/Apu/Apu.tsx [hunk 6]
```diff
@@ -0,0 +1,521 @@
...
+
+    const [apuMasterPbOn] = useSimVar('L:A32NX_OVHD_APU_MASTER_SW_PB_IS_ON', 'Bool', 500);
+    const [apuGenPbOn] = useSimVar('A:APU GENERATOR SWITCH', 'Boolean', 750);
+    const [apuAvail] = useSimVar('L:A32NX_OVHD_APU_START_PB_IS_AVAILABLE', 'Bool', 1000);
+
+    useEffect(() => {
+        if (!apuMasterPbOn && !apuAvail /* FIXME: and apu master switch is valid and apu gen push buttion is valid */) {
+            setCurrentApuGenState(apuGenState.STANDBY);
+        } else if (!(currentApuGenState === apuGenState.STANDBY) && !apuGenPbOn /* FIXME: and apu master switch valid and apu gen pb valid */) {
+            setCurrentApuGenState(apuGenState.OFF);
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Apu/Apu.tsx:80]: **issue**: The APU OFF state is not shown when expected.

image
image

File: src/instruments/src/SD/Pages/Apu/Apu.tsx [hunk 7]
```diff
@@ -0,0 +1,521 @@
...
+                && <rect className="Box" width={100} height={111} />}
+
+                <text
+                    x={50}
+                    y={20}
+                    className={`Center FontNormal
+                    ${currentApuGenState === apuGenState.OFF
+                         || (!(apuGenPotentialNormalRange && apuGenLoadNormalRange && apuGenFreqNormalRange) && currentApuGenState === apuGenState.ON) && ' Amber'}`}
+                >
+                    APU GEN
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Apu/Apu.tsx:105]: **issue**: APU GEN is shown in white when OFF, whereas amber is expected.

image
image

File: src/instruments/src/EFB/tailwind.config.js [hunk 8]
```diff
@@ -3,7 +3,7 @@
 const reactComponentsClasses = require('../../../../node_modules/@flybywiresim/react-components/build/usedCSSClasses.json');
 
 module.exports = {
-    mode: 'jit',
+    // mode: 'jit',
```
Comment (davidwalschots) [inline src/instruments/src/EFB/tailwind.config.js:6]: **question**: Is this change intentional?

**Review by davidwalschots — CHANGES_REQUESTED** [review]

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 1 above)
Comment (davidwalschots) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:79]: **suggestion**: Consider replacing the `useEffect` with the code below. Note that this doesn't necessarily fix the bug, but might make things easier to reason about.

```typescript
const isStandby = !apuMasterPbOn && !apuAvail;
const isOff = !isStandby && !apuGenPbOn;
const isOn = !isStandby && !isOff;
```

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 3 above)
Comment (ErickSharp) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:352]: What do you think would be a good memo to put here so that we know to come back and implement this specific piece of logic in the future?

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 7 above)
Comment (ErickSharp) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:105]: Thank you; addressed.

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 6 above)
Comment (ErickSharp) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:80]: Good catch; addressed.

File: src/instruments/src/EFB/tailwind.config.js (hunk 8 above)
Comment (ErickSharp) [reply src/instruments/src/EFB/tailwind.config.js:6]: No

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 1 above)
Comment (ErickSharp) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx]: > **thought**: I would remove these `FIXME` comments. As they're describing a level of detail we're not close to achieving any time soon. These comments are saying: if we want to have an even higher level of detail we need to do something here. The problem is that such comments should probably go on every other line in our code base. Let's not do that 😀.

You raise a very good point. I have gotten rid of all of the todo’s and left a couple FBWDOC comments in the components for us to reference later on when we have the means to implement what we currently can’t. :)

File: src/instruments/src/SD/Pages/Apu/Apu.tsx [hunk 9]
```diff
@@ -0,0 +1,514 @@
...
+            setApuEgtIndicationColor('Red');
+        }
+        // FBW-31-05
+    }, [apuEgt, apuEgtCaution, apuEgtWarning]);
+
+    let egtValueStyle;
+
+    if (!apuEgt.isNormalOperation()) {
+        egtValueStyle = 'Amber';
+    } else if (apuEgtIndicationColor === 'Pulse') {
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Apu/Apu.tsx:346]: **thought**: I don't see where `apuEgtIndicationColor` is ever set to pulse.

File: src/instruments/src/SD/Pages/Apu/Apu.tsx [hunk 10]
```diff
@@ -0,0 +1,514 @@
...
+            setApuEgtIndicationColor('Green');
+        }
+        if (apuEgt.value > apuEgtCaution.value && apuEgt.value <= apuEgtWarning.value) {
+            setApuEgtIndicationColor('Amber');
+        }
+        if (apuEgt.value > apuEgtWarning.value) {
+            setApuEgtIndicationColor('Red');
+        }
+        // FBW-31-05
+    }, [apuEgt, apuEgtCaution, apuEgtWarning]);
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Apu/Apu.tsx:340]: **issue**: There's no need to use `useEffect` here.
**suggestion**:

```suggestion
    // FBW-31-05
    let apuEgtIndicationColor;
    if (apuEgt.value >= apuEgtWarning.value) {
        apuEgtIndicationColor = 'Red';
    } else if (apuEgt.value >= apuEgtCaution.value) {
        apuEgtIndicationColor = 'Amber';
    } else {
        apuEgtIndicationColor = 'Green';
    }
```

File: src/instruments/src/SD/Pages/Apu/Apu.tsx [hunk 11]
```diff
@@ -0,0 +1,514 @@
...
+                    className={`FontLarger Left ${apuN.isNormalOperation() ? apuNIndicationColor : 'Amber'}`}
+                >
+                    {apuN.isNormalOperation() ? apuN.value.toFixed() : 'XX'}
+                </text>
+            </SvgGroup>
+        </>
+    );
+};
+
+const EgtGauge = ({ x, y } : ComponentPositionProps) => {
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Apu/Apu.tsx:318]: **potential issue**: EGT and N gauge display regular gauge when ECB is off. It probably show an amber line as depicted in the first image shown in the PR description. @komp1821 might be able to tell us which is correct.

Potentially faulty situation:

image

File: src/instruments/src/SD/Pages/Apu/Apu.tsx [hunk 12]
```diff
@@ -0,0 +1,514 @@
...
+        if (apuN.value >= 0 && apuN.value < 102) {
+            setApuIndicationColor('Green');
+        }
+        if (apuN.value >= 102 && apuN.value < 107) {
+            setApuIndicationColor('Amber');
+        }
+        if (apuN.value >= 107) {
+            setApuIndicationColor('Red');
+        }
+    }, [apuN]);
```
Comment (davidwalschots) [inline src/instruments/src/SD/Pages/Apu/Apu.tsx:216]: **issue**: There's no need to use `useEffect` here.
**suggestion**: See below. Note that I'm also missing handling for situations in which `apuN` is not normal.
```suggestion   
    let apuNIndicationColor;
    if (apuN.value < 102) {
        apuNIndicationColor = 'Green';
    } else if (apuN.value < 107) {
        apuNIndicationColor = 'Amber';
    } else {
        apuNIndicationColor = 'Red';
    }
```

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 11 above)
Comment (ErickSharp) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:318]: Hmm I think we will have to get more information on this because in the EIS2 information I read this doesn’t happen, and I haven’t been able to find it on any videos thus far.

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 9 above)
Comment (ErickSharp) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:346]: Hmm I think we had a chat about this a while back. Because we don’t have the ECB currently modeled, which is what signals the pulsing to occur, I think we agreed to omit it for now so it seems like this line of code is a vestige of that.

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 11 above)
Comment (ErickSharp) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:313]: > Hmm I think we will have to get more information on this because in the EIS2 information I read this doesn’t happen (or at least isn’t mentioned), and I haven’t been able to find it on any videos thus far.

Comment (ErickSharp) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:313]: > **potential issue**: EGT and N gauge display regular gauge when ECB is off. It probably show an amber line as depicted in the first image shown in the PR description. @komp1821 might be able to tell us which is correct.
> 
> Potentially faulty situation:
> 
> image

We will probably have to wait for more information from Komp like you said because from the EIS2 information regarding the way the gauge is shown that I read, this doesn’t happen, or at least isn’t mentioned.

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 9 above)
Comment (davidwalschots) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:338]: Probably removing this piece of code for now is best. Many things in the ECB are modelled, but maybe not this particular thing.

Comment (davidwalschots) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:338]: Maybe it's best to remove this remnant for now?

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 11 above)
Comment (FlyingCarpet7) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:313]: A81859CE-D203-4D9C-AC37-9EC66EFAFE80

File: src/instruments/src/SD/Pages/Apu/Apu.tsx (hunk 9 above)
Comment (ErickSharp) [reply src/instruments/src/SD/Pages/Apu/Apu.tsx:338]: Yeah I agree.

**Comment by davidwalschots** [comment]
**issue**: When the APU is ON, APU BLEED is ON, and you push the MASTER SW OFF, the BLEED VALVE indication is only partially amber. I think the valve should be fully amber with the vertical line below it in green.

image

image

**Comment by ErickSharp** [comment]
> **issue**: When the APU is ON, APU BLEED is ON, and you push the MASTER SW OFF, the BLEED VALVE indication is only partially amber. I think the valve should be fully amber with the vertical line below it in green.
> 
> image
> 
> image

Drawings I saw didn't include any color and I had to rely on the provided arrows, thanks for pointing this out : )

**Review by davidwalschots — APPROVED** [review]

**Comment by frankkopp** [comment]
Quality Assurance Tester Report

Discord: Cdr_Maverick#6475
Object of testing: #5969
Tier of Testing: 1
Date: 22.12.2021

Testing Process: 
Starting C&D / Several APU starts and shutdowns
Going through instructions
Comparing to video and FCOM

Negatives: None
(not sure how to test overloads)

Testing Results: Passed

Protocol:
<details>

- Check that 'APU GEN' is in amber if it is in OFF mode or the generator frequency is not between 390 - 410 Hz, or generator voltage is not between 110-120 V - OK
- or the generator is overloaded.  - not tested
- Generator load rises in 1V increments and is expressed in green unless it rises into overload. - OK
- Generator voltage rises in 1V increments and is expressed in green when 110 < V < 120 V or in amber otherwise. - OK (not tested >=120)
- Generator frequency rises in 1Hz increments and is expressed in green when 390 < F < 410 Hz or in amber otherwise. - OK (not tested >= 410)
- Bleed value rises in 2PSI increments. - OK8
- FUEL LO PR shown in amber in the memo region if there is low fuel pressure. . OK
- FLAP OPEN shown in green when the flap is fully open, - OK
- and in green pulsing when flap not fully closed and APU master switch push button is OFF and confirmed for 3 minutes.
- New amber mark on APU N gauge at 102% - OK
- Valve is in amber and closed when the bleed air valve is closed and the APU Bleed is selected on the overhead for a minimum of 10 seconds. - OK

</details>

**Comment by Letreivchen** [comment]
Quality Assurance Trainee Report

Discord: Letreiv#9404
Object of testing: #5969 
Tier of Testing: 1
Date : 23/12/2021

Testing Process: 
Started C&D
Carefully tested every instruction step by step in different conditions (Engines on/off, External Power on/off)

- Check that 'APU GEN' is in amber if it is in OFF mode or the generator frequency is not between 390 - 410 Hz, or generator voltage is not between 110-120 V or the generator is overloaded. ✔
- Generator load rises in 1V increments and is expressed in green unless it rises into overload.✔
- Generator voltage rises in 1V increments and is expressed in green when 110 < V < 120 V or in amber otherwise.✔
- Generator frequency rises in 1Hz increments and is expressed in green when 390 < F < 410 Hz or in amber otherwise.✔
- Bleed value rises in 2PSI increments.✔
- FUEL LO PR shown in amber in the memo region if there is low fuel pressure.✔
- FLAP OPEN shown in green when the flap is fully open.✔
- And in green pulsing when flap not fully closed and APU master switch 
push button is OFF and confirmed for 3 minutes (not tested/unable)
-New amber mark on APU N gauge at 102%✔
-Valve is in amber and closed when the bleed air valve is closed and the APU Bleed is selected on the overhead for a minimum of 10 seconds.✔



Negatives: None       
-("and in green pulsing when flap not fully closed and APU master switch push button is OFF and confirmed for 3 minutes" is an abnormal procedure, unable to test
-Load, Frequency, Volatage overloads unable to test)

Testing Results: Passed
